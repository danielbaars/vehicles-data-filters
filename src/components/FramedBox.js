import React from 'react';
import PropTypes from 'prop-types';

const FramedBox = ({ className, children }) => {
  return (
    <div className={`${className} inner`}>
      <div className={`${className}__box box`}>
        {children}
      </div>
    </div>
  );
};

export default FramedBox;

FramedBox.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
