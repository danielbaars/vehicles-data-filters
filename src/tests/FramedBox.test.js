import React from 'react';
import renderer from 'react-test-renderer';

import FramedBox from '../components/FramedBox';

describe('FramedBox', () => {
  const component = <FramedBox className='class-name'><div>Some Child</div></FramedBox>;
  it('renders and matches our snapshot', () => {
    const sComponent = renderer.create(component);
    const tree = sComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
