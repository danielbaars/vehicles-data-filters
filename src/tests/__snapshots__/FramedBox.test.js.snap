// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FramedBox renders and matches our snapshot 1`] = `
<div
  className="class-name inner"
>
  <div
    className="class-name__box box"
  >
    <div>
      Some Child
    </div>
  </div>
</div>
`;
